camp_fin/models.py

```diff
@@ -1020,113 +966,59 @@ def stack_trends(trend):
 
         cursor.execute(summed_filings, [self.id])
 
-        columns = [c[0] for c in cursor.description]
-        filing_tuple = namedtuple("Filings", columns)
-
-        summed_filings = [filing_tuple(*r) for r in cursor]
-
         balance_trend, debt_trend = [], []
 
-        if summed_filings:
-            for filing in summed_filings:
-                filing_date = filing.filed_date
-                date_array = [filing_date.year, filing_date.month, filing_date.day]
-                debts = -1 * filing.total_unpaid_debts
-                if filing.closing_balance:
-                    balance_trend.append([filing.closing_balance, *date_array])
-                    debt_trend.append([debts, *date_array])
-
-            if summed_filings[0].opening_balance:
-                first_opening_balance = summed_filings[0].opening_balance
-            else:
-                first_opening_balance = 0
-
-            if summed_filings[0].debt_carried_forward:
-                first_debt = summed_filings[0].debt_carried_forward
-            else:
-                first_debt = 0
-
-            init_date = summed_filings[0].initial_date
-
-            first_initial_date = [int(since), 1, 1]
-
-            # If the first available filing date is on or before the start date
-            # passed into this method, use that as the first date in the trendline
-            if init_date:
-                init_date_parts = [init_date.year, init_date.month, init_date.day]
-                if datetime(*init_date_parts) <= datetime(*first_initial_date):
-                    first_initial_date = init_date_parts
-
-            debt_trend.insert(0, [first_debt, *first_initial_date])
-            balance_trend.insert(0, [first_opening_balance, *first_initial_date])
+        for total_unpaid_debts, closing_balance, end_date, description in cursor:
+            period_end = {
+                "description": description,
+                "year": end_date.year,
+                "month": end_date.month,
+                "day": end_date.day,
+            }
+            balance_trend.append(
+                {
+                    "amount": closing_balance,
+                    **period_end,
+                }
+            )
+            debt_trend.append(
+                {
+                    "amount": total_unpaid_debts * -1,
+                    **period_end,
+                }
+            )
```

Comment:
Now returning lists of dictionaries, in order to pass filing description through to the view. I don't see that `trends` is used anywhere else, besides the deprecated races view.